vendor/github.com/xtaci/smux/session.go

```diff
@@ -3,36 +3,31 @@ package smux
 import (
 	"encoding/binary"
 	"io"
+	"net"
 	"sync"
 	"sync/atomic"
 	"time"
 
 	"github.com/pkg/errors"
+	siasync "github.com/NebulousLabs/Sia/sync" // TODO: Replace with github.com/NebulousLabs/trymutex
 )
 
 const (
 	defaultAcceptBacklog = 1024
 )
 
-const (
-	errBrokenPipe      = "broken pipe"
-	errInvalidProtocol = "invalid protocol version"
-	errGoAway          = "stream id overflows, should start a new connection"
+var (
+	errBrokenPipe      = errors.New("broken pipe")
+	errGoAway          = errors.New("stream id overflows, should start a new connection")
+	errInvalidProtocol = errors.New("invalid protocol version")
+	errWriteTimeout    = errors.New("unable to write to conn within the write timeout")
 )
 
-type writeRequest struct {
-	frame  Frame
-	result chan writeResult
-}
-
-type writeResult struct {
-	n   int
-	err error
-}
-
 // Session defines a multiplexed connection for streams
 type Session struct {
-	conn io.ReadWriteCloser
+	conn net.Conn
```

Comment:
for some reason the `conn` was reduced to an `io.ReadWriteCloser`, which meant I couldn't set deadlines.

Fixed now.

Reply:
probably to make it easier to mock